python/tank/context.py

```diff
@@ -117,12 +117,12 @@ def __eq__(self, other):
             return NotImplemented
 
         equal = True
-        equal &= (self.project == other.project)
-        equal &= (self.entity == other.entity)
-        equal &= (self.step == other.step)
-        equal &= (self.task == other.task)
-        equal &= (self.user == other.user)
-        equal &= (self.additional_entities == other.additional_entities)
+        equal &= (self.__project == other.__project)
```

Comment:
Sorry if i wasn't being clear about this part of the change - it is to speed up context equality. The important difference is that if you use the `self.user` property, it may reach out to shotgun to try to look up the user, which is slow. This change ensures that it is the actual *raw* context object which is being compared. It is arguable which approach is more correct, but this change will certainly speed things up in some cases.

I am not sure anything has been broken though - this was meant as an optimization only - maybe I am missing something?